hooks/tk-multi-publish2/basic/nuke/publish_files.py

```diff
@@ -0,0 +1,182 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import os
+import pprint
+import nuke
+import sgtk
+from sgtk.platform import engine
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+
+class NukePublishFilesCustomPlugin(HookBaseClass):
+    """
+    Inherits from NukePublishFilesPlugin
+    """
+    def __init__(self, parent, **kwargs):
+        """
+        Construction
+        """
+        # call base init
+        super(NukePublishFilesCustomPlugin, self).__init__(parent, **kwargs)
+
+    def _sync_frame_range(self):
+        """
+        Checks whether frame range is in sync with shotgun.
+
+        :return: True if yes false otherwise
+        """
+        # get the current engine
+        engine = sgtk.platform.current_engine()
+        # context from engine
+        context = engine.context
+        entity = context.entity
+
+        sg_entity_type = context.entity["type"]
```

Comment:
This info will only be on the Shot entity, this will fail if it is any other entity_type (i.e. Sequence/Asset), so you should catch that exception.